src/components/SampleHandModal.js

```diff
@@ -0,0 +1,129 @@
+import React, { Component } from 'react';
+
+import FoilCardImage from 'components/FoilCardImage';
+import PropTypes from 'prop-types';
+import { encodeName } from 'utils/Card';
+import { arrayShuffle } from 'utils/Util';
+
+import { Modal, ModalBody, ModalFooter, ModalHeader, Button, Col, Row, NavLink } from 'reactstrap';
+
+class SampleHandModal extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: false,
+      hand: [],
+      pool: [],
+    };
+
+    this.open = this.open.bind(this);
+    this.close = this.close.bind(this);
+    this.refresh = this.refresh.bind(this);
+    this.draw = this.draw.bind(this);
+  }
+
+  refresh() {
+    const { deck } = this.props;
+
+    const pool = [];
+    for (const col of deck) {
+      for (const card of col) {
+        pool.push(card);
+      }
+    }
+    arrayShuffle(pool);
+    const hand = pool.splice(0, 7);
+
+    this.setState({
+      hand,
+      pool,
+    });
+  }
+
+  draw() {
+    const { hand, pool } = this.state;
```

Comment:
You should make a copy of pool here.
```suggestion
    const { hand, pool } = this.state;
    const newPool = [...pool];
```